Update docs for 2021.3 release (#693)
intel/scikit-learn-intelex: doc/sources/index.rst

(`oneDAL <https://github.com/oneapi-src/oneDAL>`_). Patching scikit-learn makes it a
well-suited machine learning framework for dealing with real-life problems.

> .. image:: _static_static/scikit-learn-acceleration-2021.2.3.PNG

Reviewer: ```suggestion
.. image:: _static/scikit-learn-acceleration-2021.2.3.PNG
```

try this

For example::

    import numpy as np
    from sklearnex import patch_sklearn
    patch_sklearn()

    # Need to re-import sklearn algorithms after the patch
    from sklearn.cluster import KMeans

    X = np.array([[1,  2], [1,  4], [1,  0],
                  [10, 2], [10, 4], [10, 0]])
    kmeans = KMeans(n_clusters=2, random_state=0).fit(X)
    print(f"kmeans.labels_ = {kmeans.labels_}")

In the example above, you can see that the use of the original Scikit-learn
has not changed. This behavior is achieved through drop-in patching.

> To undo the patch Scikit-learn with::

Reviewer: ```suggestion
To undo the patch, run::
```

    sklearnex.unpatch_sklearn()

>     # Need to re-import sklearn algorithms after the unpatch

Reviewer: ```suggestion
    # You need to re-import scikit-learn algorithms after the unpatch:
```

    from sklearn.cluster import KMeans

> Patching only one algorithm can be done as follows::
> 
>     from sklearnex import patch_sklearn
>     # The names matched to scikit-learn estimators
>     patch_sklearn("SVC")
> 
> Patching several algorithm can be done as follows::
> 
>     from sklearnex import patch_sklearn
>     # The names matched to scikit-learn estimators

Reviewer: ```suggestion
You may specify which algorithms to patch:

- Patching only one algorithm::

    from sklearnex import patch_sklearn
    # The names match scikit-learn estimators
    patch_sklearn("SVC")

- Patching several algorithms::

    from sklearnex import patch_sklearn
    # The names match scikit-learn estimators
```

I'd also suggest putting this info before the part with the unpatching.
